File: LinearQueue/Queue.js
const input = require("prompt-sync");
 
class queue{
    constructor(size=5){
        this.size=size;
        this.front=0;
        this.rear=-1;
        this.array= new Array(size);
    }
    push(value){
        if(this.rear==this.size-1){
            console.log("queue is full");
            return 0;
        }
        else{
            this.rear++;
            this.array[this.rear]=value;
            console.log(value, " is added in queue");
            console.log("front: ",this.front);
            console.log("rear: ",this.rear);
            return 1;
        }
    }
    pop(){
        if(this.front>this.rear){
            console.log("queue is empty");
            return 0;
        }
        else{
            console.log(this.array[this.front], " is poped");
            this.front++;
            console.log("front: ",this.front);
            console.log("rear: ",this.rear);
            return 1;

        }
    }

    display(){
        if(this.front>this.rear){
            console.log("queue is empty");
            return 0;
        }
        else{
            for(var i=this.front;i<=this.rear;i++){
                console.log(this.array[i]);
            }
        }
    }
}


var qu =new queue();
qu.push(1);
qu.push(2);
qu.push(3);
qu.display();
qu.pop();
qu.display();
qu.push(4);
qu.push(5);
qu.pop();
qu.push(6);
qu.display();

File: LinearQueue/Queue.c
#include <stdio.h>
#include <stdlib.h>

struct queue
{int size,front,rear,*Q;};

void creatqueue(struct queue *qu){
	printf("\nenter the size of queue: ");
	scanf("%d",&qu->size);
	qu->front=0;
	qu->rear=-1;
	qu->Q=(int *)malloc(qu->size*sizeof(int));
}

void push(struct queue *qu,int value){
	if(qu->rear==qu->size-1){
		printf("\nqueue is full");
	}
	else{
		qu->rear++;
		qu->Q[qu->rear]=value;
		printf("%d id added in queue\n",value);
		printf("\nfront : %d  rear: %d\n",qu->front,qu->rear);
	}
}

int pop(struct queue *qu){
	int x=0;
	if(qu->front>qu->rear){
		printf("\nqueue is empty");
	}
	else{
		x=qu->Q[qu->front];
		printf("\n%d is poped\n", qu->Q[qu->front]);
		qu->front++;
		
	}
	return x;

}

void display(struct queue *qu){
	for(int i=qu->front;i<=qu->rear;i++){
		printf("%d\n", qu->Q[i]);
	}
}

int main()
{
	struct queue qu;
	creatqueue(&qu);
	push(&qu,10);
	push(&qu,20);
	display(&qu);
	pop(&qu);
	push(&qu,30);
	pop(&qu);
	display(&qu);
	push(&qu,40);
	push(&qu,50);
	pop(&qu);
	push(&qu,60);
	display(&qu);
	pop(&qu);
	pop(&qu);
	pop(&qu);
	return 0;
}
